alacritty_terminal/src/config/bell.rs

```diff
@@ -0,0 +1,106 @@
+use std::time::Duration;
+
+use log::error;
+use serde::{Deserialize, Deserializer};
+use serde_yaml::Value;
+
+use crate::config::{failure_default, Program, LOG_TARGET_CONFIG};
+use crate::term::color::Rgb;
+
+const DEFAULT_BELL_COLOR: Rgb = Rgb { r: 255, g: 255, b: 255 };
+
+#[serde(default)]
+#[derive(Deserialize, Clone, Debug, PartialEq, Eq)]
+pub struct BellConfig {
+    /// Visual bell animation function.
+    #[serde(deserialize_with = "failure_default")]
+    pub animation: BellAnimation,
+
+    /// Visual bell duration in milliseconds.
+    #[serde(deserialize_with = "failure_default")]
+    duration: u16,
+
+    // FIXME RGB deserialization always successedes and uses #000000 color. However it should
+    // fallback to a proper color. So it should be reworked either with
+    // https://github.com/alacritty/alacritty/pull/3507 or after that PR will land, since it'll
+    // allow to impl manual deserialization properly.
+    /// Visual bell flash color.
+    #[serde(deserialize_with = "failure_default")]
+    pub color: Rgb,
+
+    /// Command to run on bell.
+    #[serde(deserialize_with = "deserialize_command")]
+    pub command: Option<Program>,
+}
+
+impl Default for BellConfig {
+    fn default() -> Self {
+        Self {
+            animation: Default::default(),
+            duration: Default::default(),
+            color: DEFAULT_BELL_COLOR,
+            command: Default::default(),
+        }
+    }
+}
+
+impl BellConfig {
+    /// Visual bell duration in milliseconds.
+    #[inline]
+    pub fn duration(&self) -> Duration {
+        Duration::from_millis(u64::from(self.duration))
+    }
+}
+
+fn deserialize_command<'a, D>(deserializer: D) -> std::result::Result<Option<Program>, D::Error>
+where
+    D: Deserializer<'a>,
+{
+    // Deserialize to generic value.
+    let val = Value::deserialize(deserializer)?;
+
+    // Accept `None` to disable the bell command.
+    if val.as_str().filter(|v| v.to_lowercase() == "none").is_some() {
+        return Ok(None);
+    }
+
+    match <Option<Program>>::deserialize(val) {
```

Comment:
```suggestion
    match Program::deserialize(val) {
```

Reply:
The field is an option itself, so either do that or manually using `Ok(Some(command))`, so there's really not much difference, and I'd stick with the current approach, since we're using it in launcher deser already.

Reply:
I feel like that probably adds unnecessary overhead though and it doesn't clearly communicate that this cannot fail. While just using `Ok(Some(command))` makes it clear that it can never be `None` at this point.

I feel like `Program::deserialize` is the better choice here.